apache-airflow: 2.7.3 -> 2.10.5

taranarmo opened the pull request:
https://github.com/apache/airflow/releases/tag/2.10.5
Fixes #387022
The package was severely outdated and building changed a lot since 2.7.3 thus a lot of changes.
Several pinned packages required adding ignoring DeprecationWarnings, patches and skipped tests.
This also includes fix from #391551 as `aiohttp-swagger` is an upstream dependency of `connexion`.

Some providers are required now so I created a list of them and `enabledProviders` are concatenated with it.
Also the mechanism of installing providers is changed so I added a function which installs them from sources.

I tested basic functionality of the binary and fixed some bugs but proper testing required (Airflow may fail pretty unexpectedly).

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

taranarmo committed:
apache-airflow: 2.7.3 -> 2.10.5

taranarmo committed:
apache-airflow: add taranarmo to maintainers

taranarmo commented:
Current providers source code handling is a mess, convert this PR to draft until done properly

collinarnett commented:
@taranarmo Reran nixpkgs review and it LGTM!


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 396544`
Commit: `f09e0a9d294bdc8f60344e60d26cf5d3c945d472`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
  </ul>
</details>

wolfgangwalther committed:
apache-airflow: cleanup

Remove some unused arguments and use tag instead of rev in fetchers.

wolfgangwalther committed:
apache-airflow: fix build of dependencies

Some dependencies have advanced and now need to be pinned.

apache-airflow itself does not build, yet, because it needs pendulum
2.x, which requires python < 3.12, I believe.

Co-authored-by: Sergey Volkov <taranarmo@gmail.com>

wolfgangwalther committed:
apache-airflow: sort arguments and dependencies

taranarmo committed:
apache-airflow: add meta.changelog
